types/lodash/v3/index.d.ts

```diff
@@ -11242,7 +11242,7 @@ declare namespace _ {
          * @param value The value to check.
          * @return Returns true if value is a finite number, else false.
          */
-        isFinite(value?: any): boolean;
+        isFinite(value?: any): value is number;
```

Comment:
This is not correct - if `isFinite` returns false then TS will incorrectly believe that `value` is NOT a number, but it could still be a number. I think we should keep the return type as `boolean`.

Reply:
Could I know some cases where that happens?

Reply:
My quick guess would be the cases of `NaN` and `Infinity` as per MDN spec... 😅

Reply:
I think we will close this PR as it somewhat conflicts with `isNumber` function from Lodash. We just have to deal with the odd :). Thanks for your time on reviewing the PR :)

Reply:
After throughout consideration. I think we have to admit that it's better be `boolean` and client code have to cast it on their specific use cases like we did at Axon... TypeScript is just a superset of JavasScript. I will close this PR :)